lib/travis/build/script/node_js.rb

```diff
@@ -47,15 +44,47 @@ def cache_slug
 
         private
 
+          def convert_legacy_nodejs_config
+            # TODO deprecate :nodejs
+            # some old projects use language: nodejs. MK.
+            config[:node_js] = config[:nodejs]
+          end
+
+          def node_js?
+            !!config[:node_js]
+          end
+
           def version
             @version ||= begin
-              # TODO deprecate :nodejs
-              version = config[:node_js] || config[:nodejs] # some old projects use language: nodejs. MK.
+              version = config[:node_js] || 'stable'
               version = version.first if version.is_a?(Array) # it seems travis-core does not exand on nodejs anymore so we end up with an array here?
               version == 0.1 ? '0.10' : version.to_s
             end
           end
 
+          def nvm_install
+            return use_nvm_version if node_js?
```

Comment:
This doesn't read well. When I read this, I'd expect `use_nvm_version` to return a value that is usable by callers of `nvm_install`, but what we really want is just adding suitable call to `sh.cmd`.

Perhaps

```ruby
if node_js?
  use_nvm_version
else
  use_nvm_default
end
```